app/src/main/java/org/p2p/wallet/jupiter/statemanager/SwapStateManager.kt

```diff
@@ -179,29 +180,26 @@ class SwapStateManager(
         }
     }
 
-    private fun handleTokenAChange(newTokenA: SwapTokenModel): Boolean {
-        val oldTokenAZeroState = getOldTokenAZeroState(state.value) ?: return false
+    // change token A and set zero token A amount
+    private fun handleTokenAChange(newTokenA: SwapTokenModel) {
```

Comment:
I can't see if you are setting token A amount to zero. 
And do we have a requirement to make it zero?

Reply:
getOldTokenAZeroState map to SwapState.TokenAZero state
And yes if change token A then make it zero. If switch then make new tokenA(old tokenB) old tokenB amount